core/src/main/java/com/datastax/oss/driver/api/core/config/DriverConfigLoader.java

```diff
@@ -16,14 +16,123 @@
+   * basic.config-reload-interval}.
+   */
+  static DriverConfigLoader fromClasspath(String resourceBaseName) {
+    return new DefaultDriverConfigLoader(
+        () -> {
+          ConfigFactory.invalidateCaches();
+          Config config =
+              ConfigFactory.defaultOverrides()
+                  .withFallback(ConfigFactory.parseResourcesAnySyntax(resourceBaseName))
+                  .withFallback(ConfigFactory.defaultReference())
+                  .resolve();
+          return config.getConfig("datastax-java-driver");
+        });
+  }
+
+  /**
+   * Builds an instance using the driver's default implementation (based on Typesafe config), except
+   * that application-specific options are loaded from the given file.
+   *
+   * <p>More precisely, configuration properties are loaded and merged from the following
+   * (first-listed are higher priority):
+   *
+   * <ul>
+   *   <li>system properties
+   *   <li>the contents of {@code file}
+   *   <li>{@code reference.conf} (all resources on classpath with this name). In particular, this
+   *       will load the {@code reference.conf} included in the core driver JAR, that defines
+   *       default options for all mandatory options.
+   * </ul>
+   *
+   * The resulting configuration is expected to contain a {@code datastax-java-driver} section.
+   *
+   * <p>The returned loader will honor the reload interval defined by the option {@code
+   * basic.config-reload-interval}.
+   */
+  static DriverConfigLoader fromFile(File file) {
+    return new DefaultDriverConfigLoader(
+        () -> {
+          ConfigFactory.invalidateCaches();
+          Config config =
+              ConfigFactory.defaultOverrides()
+                  .withFallback(ConfigFactory.parseFileAnySyntax(file))
+                  .withFallback(ConfigFactory.defaultReference())
+                  .resolve();
+          return config.getConfig("datastax-java-driver");
+        });
+  }
+
+  /**
+   * Builds an instance using the driver's default implementation (based on Typesafe config), except
+   * that application-specific options are loaded from the given URL.
+   *
+   * <p>More precisely, configuration properties are loaded and merged from the following
+   * (first-listed are higher priority):
+   *
+   * <ul>
+   *   <li>system properties
+   *   <li>the contents of {@code url}
+   *   <li>{@code reference.conf} (all resources on classpath with this name). In particular, this
+   *       will load the {@code reference.conf} included in the core driver JAR, that defines
+   *       default options for all mandatory options.
+   * </ul>
+   *
+   * The resulting configuration is expected to contain a {@code datastax-java-driver} section.
+   *
+   * <p>The returned loader will honor the reload interval defined by the option {@code
+   * basic.config-reload-interval}.
+   */
+  static DriverConfigLoader fromUrl(URL url) {
```

Comment:
Shouldn't we have one generic method that is taking, for example, a `File`? Then methods like `fromUrl` or `fromClasspath` could only load a file using url or string and call the generic method that operates on file. That's why we will not need to have 3 almost identical methods.

Or another solution: we could create a generic method that takes a supplier function:
```
 static DriverConfigLoader from(Supplier<Config> fallbackSupplier ) {
    return new DefaultDriverConfigLoader(
        () -> {
          ConfigFactory.invalidateCaches();
          Config config =
              ConfigFactory.defaultOverrides()
                  .withFallback(fallabackSupplier)
                  .withFallback(ConfigFactory.defaultReference())
                  .resolve();
          return config.getConfig("datastax-java-driver");
        });
  }
```
And then call that method for every varian:
`from(() -> ConfigFactory.parseURL(url))`, `from(() -> ConfigFactory.parseFileAnySyntax(file)`, ...

Reply:
> a generic method that takes a supplier function

That would work if it's a private helper (we don't want to leak `Config` through the public API). The only thing is that we can't have private methods in an interface, so it would have to live somewhere else. It's just a few lines of code duplicated so I thought it was not worth it.